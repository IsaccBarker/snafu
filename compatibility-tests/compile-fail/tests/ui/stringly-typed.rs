mod and_context {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(context, other)]
    struct Error;
}

mod with_arguments {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(other(true))]
    struct Error;
}

mod missing_message {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(other)]
    struct Error;
}

mod double_message {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(other)]
    struct Error {
        message: String,
        message: String,
    }
}

mod with_context_fields {
    use snafu::Snafu;

    #[derive(Debug, Snafu)]
    #[snafu(other)]
    struct Error {
        message: String,
        user_id: i32,
    }
}

fn main() {}
