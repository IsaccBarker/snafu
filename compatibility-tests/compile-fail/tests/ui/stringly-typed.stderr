error: Cannot be both a `context` and `other` error
 --> $DIR/stringly-typed.rs:5:13
  |
5 |     #[snafu(context, other)]
  |             ^^^^^^^

error: Cannot be both a `context` and `other` error
 --> $DIR/stringly-typed.rs:5:22
  |
5 |     #[snafu(context, other)]
  |                      ^^^^^

error: expected `,`
  --> $DIR/stringly-typed.rs:13:18
   |
13 |     #[snafu(other(true))]
   |                  ^^^^^^

error: Other selectors must have a message field
  --> $DIR/stringly-typed.rs:21:5
   |
21 |     #[snafu(other)]
   |     ^^^^^^^^^^^^^^^

error: Multiple `message` attributes are not supported on a named struct
  --> $DIR/stringly-typed.rs:32:9
   |
32 |         message: String,
   |         ^^^^^^^^^^^^^^^

error: Other selectors must not have context fields
  --> $DIR/stringly-typed.rs:43:9
   |
43 |         user_id: i32,
   |         ^^^^^^^^^^^^

error[E0124]: field `message` is already declared
  --> $DIR/stringly-typed.rs:32:9
   |
31 |         message: String,
   |         --------------- `message` first declared here
32 |         message: String,
   |         ^^^^^^^^^^^^^^^ field already declared

error[E0025]: field `message` bound multiple times in the pattern
  --> $DIR/stringly-typed.rs:32:9
   |
31 |         message: String,
   |         --------------- first use of `message`
32 |         message: String,
   |         ^^^^^^^^^^^^^^^ multiple uses of `message` in pattern
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
